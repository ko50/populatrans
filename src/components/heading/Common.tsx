type Props = {
  text: string;
};

export const H1: React.FC<Props> = ({ text }) => {
  return (
    <div>
      <h1>{text}</h1>
      <span />
    </div>
  );
};
