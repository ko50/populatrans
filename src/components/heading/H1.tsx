import styles from "styles/components/heading.module.scss";

type Props = {
  text: string;
};

export const H1: React.FC<Props> = ({ text }) => {
  return (
    <div className={styles.h1}>
      <h1 className={styles.text}>{text}</h1>
      <div className={styles.underline} />
    </div>
  );
};
