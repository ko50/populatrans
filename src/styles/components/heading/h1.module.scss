@use "styles/utils/mq.scss";
@use "styles/utils/colors.scss";
@use "styles/utils/fonts.scss";

.h1 {
  width: 100%;
  padding-bottom: 2rem;
}

.text {
  color: colors.$cherry;
  font-size: 2rem;
  font-family: fonts.$kosugi;
  margin: 0;

  @include mq.mq("md") {
    font-size: 1.5rem;
  }
}

.underline {
  width: 100%;
  border-top: 2px solid colors.$stone;
}
