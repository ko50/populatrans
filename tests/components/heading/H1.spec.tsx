import React from "react";
import renderer from "react-test-renderer";
import { H1 } from "components/heading/H1";

describe("common h1 component", () => {
  it("should render correctly", () => {
    const tree = renderer.create(<H1 text="テスト" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
