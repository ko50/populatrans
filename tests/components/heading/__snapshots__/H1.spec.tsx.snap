// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common h1 component should render correctly 1`] = `
<div
  className="h1"
>
  <h1
    className="text"
  >
    テスト
  </h1>
  <div
    className="underline"
  />
</div>
`;
